sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_client_async.py

```diff
@@ -1004,11 +1044,48 @@ async def settle_messages_async(self, delivery_id: Union[int, Tuple[int, int]],
         except TypeError:
             first = delivery_id
             last = None
+
+        # Create a Message Delivery object for the Disposition
+        message_delivery = _MessageDelivery(
+            None,
+            MessageDeliveryState.WaitingToBeSent,
+            expire_time
+        )
+        on_disposition_received = partial(self._on_disposition_received_async, message_delivery)
+
         await self._link.send_disposition(
             first_delivery_id=first,
             last_delivery_id=last,
             settled=True,
             delivery_state=state,
             batchable=batchable,
-            wait=True
+            wait=True,
+            on_disposition = on_disposition_received,
         )
+
+        # Wait for the message to be settled
+        running = True
+        while running and message_delivery.state not in MESSAGE_DELIVERY_DONE_STATES:
+            # TODO: Confirm if this is the correct way to handle the do_work call
+            # calling do_work directly triggers Flow() calls which is not what we want
+            running = await self.do_work_async(flow=False)
```

Comment:
even those settlement is just Disp, we need to while loop to make sure we don't miss any of them